src/valid/analyzer.rs

```diff
@@ -169,10 +169,18 @@ enum GlobalOrArgument {
 }
 
 impl crate::Expression {
-    fn to_global_or_argument(&self) -> Result<GlobalOrArgument, ExpressionError> {
+    fn to_global_or_argument(
+        &self,
+        expression_arena: &Arena<crate::Expression>,
```

Comment:
this becomes bad with this argument. If the function input is not one expression but multiple ones, it shouldn't be a method on the `Expression` any more. Let's turn it to a freestanding function (accepting an arena and a handle)? Or put it on the arena?